it/src/test/java/com/google/cloud/teleport/it/pubsub/DefaultPubsubResourceManagerTest.java

```diff
@@ -0,0 +1,236 @@
+public final class DefaultPubsubResourceManagerTest {
+  @Rule public final MockitoRule mockito = MockitoJUnit.rule();
+
+  private static final String TEST_ID = "test-id-";
+  private static final String PROJECT_ID = "test-project";
+  private static final String TOPIC_NAME = "test-topic-name";
+  private static final String SUBSCRIPTION_NAME = "test-topic-name-sub0";
+  private static final TopicName TOPIC_REFERENCE = TopicName.of("test-project", "test-topic");
+  private static final String VALID_MESSAGE_ID = "abcdef";
+
+  @Mock private TopicAdminClient topicAdminClient;
+  @Mock private SubscriptionAdminClient subscriptionAdminClient;
+  @Mock private Topic topic;
+  @Mock private Subscription subscription;
+  @Mock private Publisher publisher;
+
+  private DefaultPubsubResourceManager testManager;
+
+  @Captor private ArgumentCaptor<TopicName> topicNameCaptor;
+  @Captor private ArgumentCaptor<SubscriptionName> subscriptionNameCaptor;
+  @Captor private ArgumentCaptor<PubsubMessage> pubsubMessageCaptor;
+
+  @Before
+  public void setUp() throws IOException {
+    testManager =
+        spy(
+            new DefaultPubsubResourceManager(
+                TEST_ID, PROJECT_ID, null, topicAdminClient, subscriptionAdminClient));
+
+    when(topic.getName()).thenReturn("projects/" + PROJECT_ID + "/topics/" + TOPIC_NAME);
+    when(subscription.getName())
+        .thenReturn("projects/" + PROJECT_ID + "/subscriptions/" + SUBSCRIPTION_NAME);
+    doReturn(publisher).when(testManager).getPublisher(any(TopicName.class));
+  }
+
+  @Test
+  public void testBuilderWithInvalidProject() {
+    IllegalArgumentException exception =
+        Assert.assertThrows(
+            IllegalArgumentException.class, () -> DefaultPubsubResourceManager.builder(""));
+    assertThat(exception).hasMessageThat().contains("projectId can not be empty");
+  }
+
+  @Test
+  public void testInvalidTopicName() {
+    IllegalArgumentException exception =
+        Assert.assertThrows(IllegalArgumentException.class, () -> testManager.getTopicName(""));
+    assertThat(exception).hasMessageThat().contains("topicName can not be empty");
+  }
+
+  @Test
+  public void testValidTopicName() {
+    TopicName topicName = testManager.getTopicName("valid-topic");
+    assertThat(topicName).isEqualTo(TopicName.of(PROJECT_ID, "test-id-valid-topic"));
+  }
+
+  @Test
+  public void testInvalidSubscriptionName() {
+    IllegalArgumentException exception =
+        Assert.assertThrows(
+            IllegalArgumentException.class, () -> testManager.getSubscriptionName(""));
+    assertThat(exception).hasMessageThat().contains("subscriptionName can not be empty");
+  }
+
+  @Test
+  public void testValidSubscriptionName() {
+    SubscriptionName subscriptionName = testManager.getSubscriptionName("valid-sub");
+    assertThat(subscriptionName).isEqualTo(SubscriptionName.of(PROJECT_ID, "test-id-valid-sub"));
+  }
+
+  @Test
+  public void testCreateTopic() {
+    // Arrange
```

Comment:
nit: You can leave out the comments if it is obvious. The comments are meant for when the blocks are large and require blank lines to keep them readable. But if the only blank lines are to separate the blocks, the comments can be left out.